content/en/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents.md

```diff
@@ -0,0 +1,121 @@
+
+There are a few ways agents may run on kubernetes cluster. Agents may run on
+nodes directly or as DaemonSets.
+
+### Why do telemetry agents talk to Docker?
+
+Historically, kubernetes was built on top of Docker. Kubernetes is managing
+networking and scheduling, Docker was placing and operating containers on a
+node. So you can get scheduling-related metadata like a pod name from Kubernetes
+and containers state information from Docker. Over time more runtimes were
+created to manage containers. Also there are projects and kubernetes features
+that generalize container status information extraction across many runtimes.
+
+Some agents are tied specifically to the Docker tool. The agents may run
+commands like [`docker ps`](https://docs.docker.com/engine/reference/commandline/ps/)
+or [`docker top`](https://docs.docker.com/engine/reference/commandline/top/) to list
+containers and processes or [docker logs](https://docs.docker.com/engine/reference/commandline/logs/)
+to subscribe on docker logs. With the deprecating of Docker as a container runtime,
+these commands will not work any longer.
+
+### Identify DaemonSets that depend on Docker {#identify-docker-dependency}
+
+If a pod wants to make calls to the `dockerd` running on the node, the pod must either:
+
+- mount the filesystem containing the Docker daemon's privileged socket, as a
+  {{< glossary_tooltip text="volume" term_id="volume" >}}; or
+- mount the specific path of the Docker daemon's privileged socket directly, also as a volume.
+
+For example: on COS images, Docker exposes its Unix domain socket at
+`/var/run/docker.sock` This means that the pod spec will include a
+`hostPath` volume mount of `/var/run/docker.sock`.
+
+Here's a sample shell script to find Pods that have a mount directly mapping the
+Docker socket. This script outputs the namespace and name of the pod. You can
+remove the grep `/var/run/docker.sock` to review other mounts.
+
+```bash
+kubectl get pods --all-namespaces \
+-o=jsonpath='{range .items[*]}{"\n"}{.metadata.namespace}{":\t"}{.metadata.name}{":\t"}{range .spec.volumes[*]}{.hostPath.path}{", "}{end}{end}' \
+| sort \
+| grep '/var/run/docker.sock'
+```
+
+{{< note >}}
+There are alternative ways for a pod to access Docker on the host. For instance, the parent
+directory `/var/run` may be mounted instead of the full path (like in [this
+example](https://gist.github.com/itaysk/7bc3e56d69c4d72a549286d98fd557dd)).
+The script above only detects the most common uses.
+{{< /note >}}
+
+### Detecting Docker dependency from node agents
+
+In case your cluster nodes are customized and install additional security and
+telemetry agents on the node, make sure to check with the vendor of the agent whether it has dependency on Docker.
+
+## Telemetry and security agent vendors
+
+This section is intended to collect information about various telemetry and
+security agents that may have dependency on containers runtime.
+
+### SignalFx (Splunk)
```

Comment:
```suggestion
### SignalFx (Splunk) {#vendor-signalfx-splunk}
```
@SergeyKanzhelev this seems to be outside what the [content guide](https://kubernetes.io/docs/contribute/style/content-guide/) permits. I'm going to
/hold
this PR for now; we can unhold it if it becomes clear that these vendor-specific details are OK to add.

Reply:
agree! As I said, one blocker for some user is to transition monitoring agents to the new runtime. So it will be super helpful to either accumulate this information here or has a directory of links for easy discoverability.